apps/api/src/server/applications/state-machine/__tests__/document.machine.test.js

```diff
@@ -0,0 +1,32 @@
+import test from 'ava';
+import { transitionState } from '../../../utils/transition-state.js';
+/**
+ * @param {object} t
+ * @param {{initialState: string, action: string, expectedState: string, context: object, hasChanged: boolean}} config
+ */
+function applyAction(t, { initialState, action, expectedState, context, hasChanged }) {
+	const nextState = transitionState({
+		caseType: 'document',
+		context,
+		status: initialState,
+		machineAction: action
+	});
+
+	t.deepEqual(nextState.value, expectedState);
+	t.is(nextState.changed, hasChanged);
+}
+
+applyAction.title = (providedTitle, { initialState, action, expectedState, context, hasChanged }) =>
+	`Document State Machine: ${providedTitle}: from state [${JSON.stringify(initialState)}]
+	with context ${JSON.stringify(context)} action [${action}] produces state
+	[${JSON.stringify(expectedState)}] ${hasChanged ? '' : ' without'} having transitioned`;
+
+for (const parameter of [['awaiting_upload', 'uploading', 'not_yet_checked', {}, true]]) {
```

Comment:
We are only testing a single scenario here, could we please add all the allowed and disallowed transition to these tests?

Reply:
would it look smth like this: 

```
for (const parameter of [['uploading', 'not_yet_checked', 'failed_checks', {}, true]]) {
	test(applyAction, {
		initialState: parameter[0],
		action: parameter[1],
		expectedState: parameter[2],
		context: parameter[3],
		hasChanged: parameter[4]
	});
}
```

Reply:
hey sorry didn't see this message earlier. It would be a case of adding more items to the array against which we are iteratively running the tests:
```
for (const parameter of [['awaiting_upload', 'uploading', 'not_yet_checked', {}, true], ['uploading', 'not_yet_checked', 'failed_checks', {}, true]]) {
```

please refer the other state machine tests for the exact formatting

Reply:
i changed it - is this okay?